hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -822,7 +822,30 @@ private <S, R> void onLocateComplete(Function<RpcChannel, S> stubMaker,
         } else {
           callback.onRegionComplete(region, r);
         }
-        if (unfinishedRequest.decrementAndGet() == 0 && locateFinished.get()) {
+
+        boolean complete = unfinishedRequest.decrementAndGet() == 0 && locateFinished.get();
+
+        if (e == null && r != null) {
+          ServiceCaller<S, R> updatedCallable = callback.getNextCallable(r, region);
+          if (updatedCallable != null) {
+            // We are launching a new request now, so un-set complete
+            complete = false;
+            long waitIntervalMs = callback.getWaitIntervalMs(r, region);
+            LOG.trace("Coprocessor returned incomplete result. "
+              + "Sleeping for {} millis before making follow-up request.", waitIntervalMs);
+            if (waitIntervalMs > 0) {
+              AsyncConnectionImpl.RETRY_TIMER.newTimeout(
+                (timeout) -> onLocateComplete(stubMaker, updatedCallable, callback, endKey,
+                  endKeyInclusive, locateFinished, unfinishedRequest, loc, null),
+                waitIntervalMs, TimeUnit.MILLISECONDS);
+            } else {
+              onLocateComplete(stubMaker, updatedCallable, callback, endKey, endKeyInclusive,
```

Comment:
I think we'd better do another abstraction on the send request part, since here we just send a new request to the same region, we do not need to determine whether locate is finished?

Reply:
Sure, this is refactored now